src/main/java/com/nike/cerberus/jobs/HystrixMetricsProcessingJob.java

```diff
@@ -0,0 +1,161 @@
+
+    private static final Logger log = LoggerFactory.getLogger(HystrixMetricsProcessingJob.class);
+
+    private final CodahaleMetricsCollector metricsCollector;
+
+    @Inject
+    public HystrixMetricsProcessingJob(CodahaleMetricsCollector metricsCollector) {
+        this.metricsCollector = metricsCollector;
+    }
+
+    @Override
+    public void doRun() throws JobInterruptException {
+        log.debug("Running hystrix metrics processing job");
+        try {
+            processHystrixCommandMetrics();
+            processHystrixThreadPoolMetrics();
+        } catch (JobInterruptException e) {
+            throw e;
+        } catch (Exception e) {
+            log.warn("Error processing Hystrix metrics", e);
+        }
+    }
+
+    public void processHystrixCommandMetrics() {
+        for (HystrixCommandMetrics metrics : HystrixCommandMetrics.getInstances()) {
+            boolean isCircuitOpen = HystrixCircuitBreaker.Factory.getInstance(metrics.getCommandKey()).isOpen();
+
+            log.debug("group:{}, commandKey:{}, CircuitOpen:{}, Mean:{}, 95%:{}, 99%:{}, 99.5%:{}, {}",
+                    metrics.getCommandGroup().name(),
+                    metrics.getCommandKey().name(),
+                    isCircuitOpen,
+                    metrics.getExecutionTimeMean(),
+                    metrics.getExecutionTimePercentile(95.0),
+                    metrics.getExecutionTimePercentile(99.0),
+                    metrics.getExecutionTimePercentile(99.5),
+                    metrics.getHealthCounts()
+            );
+
+            String baseMetricName = "cerberus.hystrix.command" +
+                    metrics.getCommandGroup().name().toLowerCase() + "." +
+                    metrics.getCommandKey().name().toLowerCase();
+
+            getOrCreateSettableGauge(baseMetricName + ".circuitOpen").ifPresent((gauge) ->
+                    gauge.setValue(isCircuitOpen ? 1 : 0));
+
+            getOrCreateSettableGauge(baseMetricName + ".mean").ifPresent((gauge) ->
+                    gauge.setValue(metrics.getExecutionTimeMean()));
+
+            getOrCreateSettableGauge(baseMetricName + ".95th").ifPresent((gauge) ->
+                    gauge.setValue(metrics.getExecutionTimePercentile(95.0)));
+
+            getOrCreateSettableGauge(baseMetricName + ".99th").ifPresent((gauge) ->
+                    gauge.setValue(metrics.getExecutionTimePercentile(99.0)));
+
+            getOrCreateSettableGauge(baseMetricName + ".995th").ifPresent((gauge) ->
+                    gauge.setValue(metrics.getExecutionTimePercentile(99.5)));
+
+            getOrCreateSettableGauge(baseMetricName + ".totalCount").ifPresent((gauge) ->
+                    gauge.setValue(metrics.getHealthCounts().getTotalRequests()));
+
+            getOrCreateSettableGauge(baseMetricName + ".errorCount").ifPresent((gauge) ->
+                    gauge.setValue(metrics.getHealthCounts().getErrorCount()));
+        }
+    }
+
```

Comment:
Did we check with anyone on an existing library or sample code we can use?

Reply:
We asked @nicmunroe some initial questions. This is an adaptation from his response.